### PR TITLE
2105 use leaflet

### PR DESCRIPTION
**Pull request by robkooper**
Replace google maps with leaflet. This speeds up rendering of many sites, and clusters markers together. Clicking on a cluster will zoom into that region.

Redid most of the create site code, right click on the map will now open a new page to enter the site information.

For an example of this see https://pecan-docker.ncsa.illinois.edu/pecan/02-modelsite.php

## Motivation and Context
google maps could now cost money depending on the number of hits.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [X] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ For more information about this file see also [Keep a Changelog](http://keepacha
 ## Unreleased
 
 ### Fixes
+- Removed google maps and replaced with leaflet #2105
```
Comment (mdietze) [inline CHANGELOG.md]: would also mention the add site functionality

File: web/02a-createsite.php [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    }
+    if ($("#lon").val() == "") {
+      $("#error").html("Enter lon to continue");
+      return false;
+    }
+    // if ($("#elevation").val() == "") {
+    //   $("#error").html("Enter elevation to continue");
+    //   return false;
+    // }
+    if ($("#temperature").val() == "") {
```
Comment (mdietze) [inline web/02a-createsite.php]: The current Add Site interface requires a whole lot of information that's optional (temp, precip, clay, sand, county). Also, the interface doesn't give any hints about which fields are required and which are optional. I'd recommend adding a * or BOLD to the required variables.

File: web/02a-createsite.php [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      <div class="spacer"></div>
+    </form>
+<?php left_footer(); ?>    
+  </div>
+  <div id="output">
+    <form id="site">
+      <fieldset>
+        <legend>Site Information</legend>
+
+        <label>Site name:</label>
```
Comment (mdietze) [inline web/02a-createsite.php]:
```suggestion
        <label>Site name (*required):</label>
```

File: web/02a-createsite.php [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+        <label>Country:</label>
+        <input id="country" size="30" type="text" name="country"></input>
+        <div class="spacer"></div>
+
+        <label>Timezone:</label>
+        <input id="timezone" size="30" type="text" name="timezone"></input>
+        <div class="spacer"></div>
+
+        <label>Latitude:</label>
```
Comment (mdietze) [inline web/02a-createsite.php]:
```suggestion
        <label>Latitude (*required):</label>
```

File: web/02a-createsite.php [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        <input id="lat" size="30" type="text" name="lat" value="<?php echo $lat; ?>"></input>
+        <div class="spacer"></div>
+
+        <label>Longitude:</label>
```
Comment (mdietze) [inline web/02a-createsite.php]:
```suggestion
        <label>Longitude (*required):</label>
```

File: CHANGELOG.md (hunk 1 above)
Comment (robkooper) [reply CHANGELOG.md]: done

File: web/02a-createsite.php (hunk 2 above)
Comment (robkooper) [reply web/02a-createsite.php]: only sitename, lat, lon is now required.

**Pull request merged**